service/history/execution/context_util.go

```diff
@@ -21,15 +21,44 @@
 package execution
 
 import (
+	"strconv"
 	"time"
 
+	"github.com/uber/cadence/service/history/config"
+
 	"github.com/uber/cadence/common/log"
 	"github.com/uber/cadence/common/log/tag"
 	"github.com/uber/cadence/common/metrics"
 	"github.com/uber/cadence/common/persistence"
 	"github.com/uber/cadence/common/types"
 )
 
+func emitLargeWorkflowShardIDStats(logger log.Logger, metricsClient metrics.Client, config *config.Config, shardID int,
+	domainName string, workflowID string, currentHistorySize int64, currentHistoryCount int64, oldHistorySize int64, oldHistoryCount int64, blobSize int64) {
```

Comment:
too many same-typed args for safety tbh.  make a temp struct, so it's essentially not possible to screw up the arg ordering?  it might also help the calling code be clearer.

also, since this is only used by the context and it has 5 args that come directly from the context: it'd probably be a lot simpler to just make this a method, so it can get those values from itself rather than as a bunch of arguments.  shouldn't change tests much either, since you can construct a minimal `contextImpl` pretty easily.

Reply:
extending the history-stats struct like you've done makes sense at a glance.  checking more thoroughly, but I think I like it - seems like it's all for the same purpose.

Reply:
neeeever mind - that's a persistence object.  that deserves more scrutiny.
particularly with the "old" values - passing those as an argument might be better, rather than saving them to the database.

Reply:
ohh those get saved? I'm guessing somewhere along the way it saves the new context? I'll just pass them then. Managed to cut it down to only 3 fields now though rest are just "c.blah"

Reply:
yea, I think all these need to be separate.  current count is stored in the database and this func relies on it to raise the count by the num of new events, and that ends up being written back to the database.

all the new metrics don't need that.  they're inferred from other recorded values and are just being held temporarily in memory to emit monitoring.  no need to duplicate them in the DB / risk them getting out of sync.

Reply:
yeah, after turning it into a method I think it's small enough that it's less risky to have those args.  lots just have a bad track record of leading to accidentally-incorrect ordering.